code-challenges.js

```diff
@@ -13,22 +28,89 @@ var fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
 var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+const odd = (array) => {
+
+        // tools: .filter === number
+        let filteredArray = array.filter((value)=> typeof (value) == 'number')
+        console.log('filteredArray', filteredArray);
+        //.filter !== even
+        let oddArray = filteredArray.filter((value)=> value % 2 !== 0)
+        console.log('oddArray', oddArray);
+        //  numbers.sort((a, b) => a - b);
+        let oddSorted= oddArray.sort((a, b) => a - b);
+        return oddSorted
+}
+console.log(odd(fullArr1));
+console.log(odd(fullArr2));
+
 
 
 
 // --------------------3) Create a function that takes in a string of a single word and returns the middle letter of the word. If the word is an even number of letters, return the two middle letters.
 
 var middleLetters1 = "hello"
 // Expected output: “l”
+
 var middleLetters2 = "rhinoceros"
 // Expected output: “oc”
 
+const middle = (string) => {
+  //tools string.split
+  let stringSplit = string.split("")
+  console.log('stringSplit', stringSplit);
+  // lenght %2 (even or odd)
+  let lengthNum = stringSplit.length
+  console.log('lengthNum', lengthNum);
+  let middleNum = lengthNum % 2
+  console.log('middleNum', middleNum);
+
+  //if %2 =1 in which case return value of the index of num + .5
+  if (middleNum === 1){
+    let middleIndex = lengthNum / 2 + .5
+    console.log(middleIndex);
+    return stringSplit[middleIndex]
+  }
+  // if %2 =0 return value of index of ^variable^ and value of index +1 in a string (.join)
+  else if (middleNum === 0){
+    let middleIndex1 = lengthNum/2- 1
+    let middleIndex2 =lengthNum/2
+    console.log(middleIndex1, middleIndex2);
+    return stringSplit[middleIndex1] + stringSplit[middleIndex2]
+  }
+}
+
+console.log(middle(middleLetters1));
+console.log(middle(middleLetters2));
```

Comment:
this is great! I would comment out the console.logs though just so your returns are clearer :)